src/commands/initProjectForVSCode/verifyVSCodeConfig.ts

```diff
@@ -138,37 +140,136 @@ async function verifyJavaDeployConfigIsValid(projectLanguage: ProjectLanguage |
-        actionContext.properties.javaDeployConfigValid = 'false';
+        actionContext.properties.verifyConfigPrompt = 'updateJavaDeployConfig';
 
         const settingKey: string = 'showJavaDeployConfigWarning';
         const message: string = localize('updateJavaDeployConfig', 'Your deploy configuration is out of date and may not work with the latest version of the Azure Functions extension for VS Code.');
         const learnMoreLink: string = 'https://aka.ms/AA41zno';
-        if (await promptToUpdateProject(workspacePath, settingKey, message, learnMoreLink)) {
+        if (await promptToUpdateProject(workspacePath, settingKey, message, learnMoreLink, actionContext)) {
             actionContext.suppressErrorDisplay = false;
             await initProjectForVSCode(actionContext, workspacePath, projectLanguage);
-            actionContext.properties.updatedJavaDebugConfig = 'true';
         }
     }
 }
 
-async function promptToUpdateProject(fsPath: string, settingKey: string, message: string, learnMoreLink: string): Promise<boolean> {
+async function promptToUpdateProject(fsPath: string, settingKey: string, message: string, learnMoreLink: string, actionContext: IActionContext): Promise<boolean> {
     if (getFuncExtensionSetting<boolean>(settingKey)) {
         const updateConfig: vscode.MessageItem = { title: localize('reinit', 'Reinitialize Project') };
         const result: vscode.MessageItem = await ext.ui.showWarningMessage(message, { learnMoreLink }, updateConfig, DialogResponses.dontWarnAgain);
         if (result === DialogResponses.dontWarnAgain) {
+            actionContext.properties.verifyConfigResult = 'dontWarnAgain';
             await updateWorkspaceSetting(settingKey, false, fsPath);
         } else {
+            actionContext.properties.verifyConfigResult = 'update';
             return true;
         }
     }
 
     return false;
 }
 
+async function verifyTargetFramework(projectLanguage: ProjectLanguage, workspacePath: string, projectPath: string, actionContext: IActionContext): Promise<void> {
+    const settingKey: string = 'showTargetFrameworkWarning';
+    if (getFuncExtensionSetting<boolean>(settingKey)) {
+
+        const projFileName: string | undefined = projectLanguage === ProjectLanguage.CSharp ? await tryGetCsprojFile(projectPath) : await tryGetFsprojFile(projectPath);
+        if (projFileName) {
+
+            const targetFramework: string | undefined = await tryGetTargetFramework(path.join(projectPath, projFileName));
+            if (targetFramework) {
+
+                const tasksResult: IVerifyFrameworkResult | undefined = verifyTasksFramework(workspacePath, targetFramework);
+                const settingsResult: IVerifyFrameworkResult | undefined = verifySettingsFramework(workspacePath, targetFramework);
+
+                const mismatchTargetFramework: string | undefined = (tasksResult && tasksResult.mismatchTargetFramework) || (settingsResult && settingsResult.mismatchTargetFramework);
```

Comment:
Will there ever be a case where `tasksResult.mismatchTargetFramework` and `settingsResult.mismatchTargetFramework` are both defined?